DSCResources/MSFT_xIPAddress/MSFT_xIPAddress.psm1

```diff
@@ -134,48 +130,162 @@ function ValidateProperties
+            "in the configuration and try again"
+            ) -join "" | Out-String
+        )
     }
 
     try
     {
         Write-Verbose -Message "Checking the IPAddress ..."
         #Get the current IP Address based on the parameters given.
-        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily $AddressFamily -ErrorAction Stop
+        $currentIP = Get-NetIPAddress -InterfaceAlias $InterfaceAlias -AddressFamily `
+            $AddressFamily -ErrorAction Stop
+
+        # Build hash table of current settings
+        $currentSettings = @{}
+        $currentSettings['IPAddress'] = $currentIP.IPAddress
+        $currentSettings['PrefixLength'] = $currentIP.PrefixLength
+        $currentSettings['InterfaceAlias'] = $currentIP.InterfaceAlias
+
+        # Flag to signal whether settings are correct
+        $requiresChanges = $false
 
         #Test if the IP Address passed is equal to the current ip address
-        if(!$currentIP.IPAddress.Contains($IPAddress))
+        if(!$currentSettings['IPAddress'].Equals($IPAddress))
         {
-            Write-Verbose -Message "IPAddress not correct. Expected $IPAddress, actual $($currentIP.IPAddress)"
-            $Parameters = @{}
+            Write-Verbose -Message ( @(
+                "IPAddress does NOT match desired state. Expected $IPAddress, actual "
+                "$($currentSettings['IPAddress'])."
+                ) -join "" | Out-String
+            )
+            $requiresChanges = $true
+        }
+        else
+        {
+            Write-Verbose -Message "IPAddress is correct."
+        }
+
+        #Test if the Subnet Mask passed is equal to the current subnet mask
+        if(!$currentSettings['PrefixLength'].Equals([byte]$SubnetMask))
+        {
+            Write-Verbose -Message ( @(
+                "Subnet mask does NOT match desired state. Expected $SubnetMask, actual "
+                "$($currentSettings['PrefixLength'])."
+                ) -join "" | Out-String
+            )
+            $requiresChanges = $true
+        }
+        else
+        {
+            Write-Verbose -Message "Subnet mask is correct."
+        }
 
-            #Apply is true in the case of set - target resource - in which case, it will set the new IP Address
+        #Test if the Default Gateway passed is equal to the current default gateway
+        if($DefaultGateway)
+        {
+            $gateway = $DefaultGateway
+            if(!([System.Net.Ipaddress]::TryParse($gateway, [ref]0)))
+            {
+                throw ( @(
+                    "Default Gateway *$DefaultGateway* is NOT in the correct format. Please "
+                    "correct the gateway ipaddress in the configuration and try again"
+                    ) -join "" | Out-String
+                )
+            }
+
+            $netIpConf = Get-NetIPConfiguration -InterfaceAlias $InterfaceAlias -ErrorAction Stop
+            $currentSettings['DefaultGateway'] = (
```

Comment:
I found this really hard to read what is happening with this.  It took a bit of thinking to realize what properties you were calling and what you were ultimately after.

I'd suggest rewriting 199-202 to look like the following

````
$netIpConfig = Get-NetIPConfiguration -InterfaceAlias $InterfaceAlias -ErrorAction Stop

# IPv6DefaultGateway
# IPv4DefaultGateway
$defaultGatewayProperty = "$($AddressFamily)DefaultGateway"
$currentSettings['DefaultGateway'] = $($netIpConfig.$defaultGatewayProperty).NextHop
````

This will have some comments on what you're creating int the string, and means you are not breaking up the assignment to multiple lines which made it a bit more challenging to understand.